### DISCUSSION TITLE
Change QueryResult.row and QueryResultRow public to make impl TryGetable for custom type possible

### DISCUSSION THREAD

**Discussion by deeper-thought** [SeaQL/sea-orm#2454] [Ideas]
When try to add a new type mapping from a database datatype which is not included in sea-orm. I should impl sea_orm::TryGetable for my custom type.

```
impl sea_orm::TryGetable for Vector {
    fn try_get_by<I: sea_orm::ColIdx>(res: &QueryResult,idx: I) -> Result<Self, sea_orm::TryGetError> {
        match &res.row {  // <-- row is private
            ...
        }
    }
}
```

But QueryResult.row is private and there is no func to get raw result. Then I cannot transform from raw result to my type.
So QueryResult.row and enum QueryResultRow should be pub.

**Comment by tyt2y3** [comment]
Hi, are you implementing pg Vector support for SeaORM? If so it's better for you to implement it inside SeaORM and open a PR
